src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -0,0 +1,194 @@
+                }
+                else
+                {
+                    _logger.Verbose("User is not specified for silent token request");
+                }
+
+                if (resultEx != null && !string.IsNullOrEmpty(resultEx.AccessToken))
+                {
+                    _logger.Verbose("Token is returned from silent call");
+                    readyForResponse.Release();
+                    return;
+                }
+                else
+                {
+                    _logger.Verbose("Token is not returned from silent backgroud call");
+                }
+
+                _logger.Verbose("Starting Authentication Activity");
+
+                if (resultEx == null)
+                {
+                    _logger.Verbose("Initial request to broker");
+                    // Log the initial request but not force a prompt
+                }
+
+                // onActivityResult will receive the response for this activity.
+                // Lauching this activity will switch to the broker app.
+                Intent brokerIntent = _brokerHelper.GetIntentForInteractiveBrokerRequest(brokerPayload, _activity);
+                if (brokerIntent != null)
+                {
+                    try
+                    {
+                        _logger.Verbose(
+                            "Calling activity pid:" + AndroidNative.OS.Process.MyPid()
+                            + " tid:" + AndroidNative.OS.Process.MyTid() + "uid:"
+                            + AndroidNative.OS.Process.MyUid());
+
+                        _activity.StartActivityForResult(brokerIntent, 1001);
+                    }
+                    catch (ActivityNotFoundException e)
+                    {
+                        _logger.ErrorPii(e);
+                    }
+                }
+        }
+            else
+            {
+                throw new MsalException(MsalError.NoBrokerAccountFound, "Please add the selected account to the broker");
+            }
+
+            await readyForResponse.WaitAsync().ConfigureAwait(false);
+        }
+
+        internal static void SetBrokerResult(Intent data, int resultCode)
+        {
+            if (data == null)
+            {
+                readyForResponse.Release();
+                return;
+            }
+
+            if (resultCode != BrokerResponseCode.ResponseReceived)
+            {
+                resultEx = new MsalTokenResponse
+                {
+                    Error = data.GetStringExtra(BrokerConstants.ResponseErrorCode),
+                    ErrorDescription = data.GetStringExtra(BrokerConstants.ResponseErrorMessage),
+                };
+            }
+            else
+            {
+                resultEx = CreateMsalTokenResponseFromResult(data.GetStringExtra(BrokerConstants.BrokerResultV2));
+            }
+
+            readyForResponse.Release();
+        }
+
+        private static MsalTokenResponse CreateMsalTokenResponseFromResult(string brokerResult)
```

Comment:
This class generally looks ok but we need to get rid of all the statics, maybe with the exception of the semaphore, which I assume you're using to prevent more than 1 request to the broker at the same time.